tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py

```diff
@@ -91,36 +91,25 @@ def setup_rhsm(shell):
         ).returncode
         == 0
     )
-    if SYSTEM_RELEASE_ENV in ("centos-7", "oracle-7"):
-        assert (
-            shell(
-                f"curl -o /etc/yum.repos.d/client-tools.repo https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-7-server.repo \
-                    --proxy http://{TEST_VARS['PROXY_SERVER']}:{TEST_VARS['PROXY_PORT']}",
-                silent=True,
-            ).returncode
-            == 0
-        )
-    else:
-        assert (
-            shell(
-                f"curl -o /etc/yum.repos.d/client-tools.repo https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-8.repo \
-                    --proxy http://{TEST_VARS['PROXY_SERVER']}:{TEST_VARS['PROXY_PORT']}",
-                silent=True,
-            ).returncode
-            == 0
-        )
-        # On CentOS 8.5 we need to replace the $releasever in the url to 8.5,
-        # otherwise the dnf will complain with dependency issues.
-        if "centos-8" in SYSTEM_RELEASE_ENV:
-            shell(r"sed -i 's#\$releasever#8.5#' /etc/yum.repos.d/client-tools.repo")
-
-    # On Oracle Linux 7 a "rhn-client-tools" package may be present on
-    # the system which prevents "subscription-manager" to be installed.
-    if "oracle" in SYSTEM_RELEASE_ENV:
-        shell("yum remove rhn-client-tools -y")
-        shell("echo 'exclude=rhn-client*' >> /etc/yum.conf")
-
-    shell("yum -y install subscription-manager subscription-manager-rhsm-certificates")
+
+    client_tools_repo = "client-tools-for-rhel-7-server.repo"
+    major_version = SystemInformationRelease.version.major
+    if major_version in (8, 9):
+        client_tools_repo = f"client-tools-for-rhel-{major_version}.repo"
+
+    ct_repo_shell_call = f"curl -o /etc/yum.repos.d/client-tools.repo https://cdn-public.redhat.com/content/public/repofiles/{client_tools_repo} \
+                    --proxy http://{TEST_VARS['PROXY_SERVER']}:{TEST_VARS['PROXY_PORT']}"
+
+    assert shell(ct_repo_shell_call, silent=True).returncode == 0
+
+    # On CentOS 8.5 we need to replace the $releasever in the url to 8.5,
+    # otherwise the dnf will complain with dependency issues.
+    if "centos-8" in SYSTEM_RELEASE_ENV:
+        shell(r"sed -i 's#\$releasever#8.5#' /etc/yum.repos.d/client-tools.repo")
+
```

Comment:
For this we can use the newly created class SubscriptionManager() that you introduced in this PR

Reply:
If it doesn't work through the proxy, I blame you :sweat_smile:

Reply:
<img src="https://media1.giphy.com/media/a93jwI0wkWTQs/giphy.gif"/>